### PR TITLE
[Android] LoadApplication should not override renderers in derived activity

### PR DESCRIPTION
**Pull request by adrianknight89**
### Description of Change

If you inherit from `FormsAppCompatActivity` and call `LoadApplication`, any default renderers you may have set will be overridden by XF defaults. This should expose a way to bypass this step.
### Bugs Fixed
- https://bugzilla.xamarin.com/show_bug.cgi?id=42818
- https://bugzilla.xamarin.com/show_bug.cgi?id=38117
### API Changes

Added:
- protected virtual bool ShouldUseDefaultHandlersForRenderers => true;
- protected virtual void RegisterAllDefaultHandlersForRenderers()
### PR Checklist
- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [ ] Consolidate commits as makes sense

### CODE REVIEW

**Comment by adrianknight89** [comment]
The idea behind setting `LoadApplication` virtual was that handlers would be defined in the derived method and base would be invoked afterwards. I think I'm going to create `RegisterHandlerForDefaultRenderers` and mark it virtual so that `LoadApplication` doesn't have to be virtual and base can be invoked at the top of the method.

File: Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs [hunk 1]
```diff
@@ -105,25 +107,13 @@ public void SetStatusBarColor(AColor color)
...
-				_renderersAdded = true;
-			}
+			RegisterAllDefaultHandlersForRenderers();
 
 			if (application == null)
-				throw new ArgumentNullException("application");
+				throw new ArgumentNullException(nameof(application));
 
 			_application = application;
-			(application as IApplicationController)?.SetAppIndexingProvider(new AndroidAppIndexProvider(this));
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs]: `?` is unnecessary.

**Comment by jassmith** [comment]
The issue seems to be that somehow the new override renderers are clobbering peoples custom renderers.

**Comment by adrianknight89** [comment]
@jassmith Anything happening with this PR?

**Comment by hartez** [comment]
These bugs were fixed by [PR 122](https://github.com/xamarin/Xamarin.Forms/pull/112); custom renderers registered for default controls are no longer clobbered by FormsAppCompatActivity on startup.

**Pull request closed without merging**